unit_test.py

```diff
@@ -0,0 +1,1371 @@
+    shuffle(permission_list)
+    return [u.get_power(p) for p in
+            permission_list[0:randint(1, len(permission_list))]]
+
+
+# Generate a random value for role_name attribute in User class
+def rand_role(roles=['SuperAdmin', 'Admin', 'Student']):
+    return roles[randint(0, len(roles)-1)]
+
+
+# Generate a random value for role_name attribute in User class
+# among all the possible admin role names
+def rand_admin(roles=['SuperAdmin', 'Admin']):
+    return roles[randint(0, len(roles)-1)]
+
+
+# Generate a random integer which is the number of a loop in many testing
+# functions
+def rand_round(min_round=1, max_round=5):
+    return randint(min_round, max_round)
+
+
+# Generate a random list of students in string format with comma as delimitor
+def rand_student_names(number=5):
+    s = ''
+    for _ in range(number):
+        username = randlength_word()
+        # Avoid repetition of username
+        while s.find(username) != -1:
+            username = randlength_word()
+        s += (username+',')
+    s = s.rstrip(',')
+    return s
+
+
+class TestCase(unittest.TestCase):
+    # this is automatically called for us when we run the test
+    def setUp(self):
+        # It disables the error catching during request handling so that we get
+        # better error reports when performing test requests against the
+        # application.
+        app.config['TESTING'] = True
+        # The database used for this suit of tests is not the one used by the
+        # app. Before we run this test, we need to create a local database
+        # called lemur_test
+        app.config['SQLALCHEMY_DATABASE_URI'] = 'postgres://localhost/lemur_test'
+        self.app = app.test_client()
+        # Create all the tables
+        db.create_all()
+        # Insert all the roles(Student, Admin, SuperAdmin) that will be used
+        m.Role.insert_roles()
+        # Set a random seed according to current time
+        seed(datetime.now())
+
+    # tidy up after a test has been run
+    def tearDown(self):
+        # remove and clean the database completely
+        ds.remove()
+        db.drop_all()
+
+    # ---- Other helper functions ---
+    # Log in the user and redirect to a corresponding page
+    def login(self, username):
+        return self.app.post('/login', data=dict(
+               username=username), follow_redirects=True)
+
+    # Log out the user and redirect to a corresponding page
+    def logout(self):
+        return self.app.get('/logout', follow_redirects=True)
+
+    # Decode a response object and return its data in the beutiful soup format
+    def decode(self, response, format='utf-8'):
+        html = response.data.decode('utf-8')
+        soup = BeautifulSoup(html, 'html5lib')
+        return soup
+
+    # Generate a json format dictionary that consists of information needed to
+    # create a new lab with random values
+    def construct_lab_json(self):
+        lab_json = {'labName': randlength_word(),
```

Comment:
python dictionaries are not json. This method returns a python dictionary, not a json object.